Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Image/StreamingImageAsset.h

```diff
@@ -85,6 +86,19 @@ namespace AZ
             //! Get image data for specified mip and slice. It may return empty array if its mipchain assets are not loaded
             AZStd::array_view<uint8_t> GetSubImageData(uint32_t mip, uint32_t slice);
 
+            //! Get single image pixel value for specified mip and slice
+            template<typename T>
+            T GetSubImagePixelValue(uint32_t x, uint32_t y, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
+
+            //! Retrieve a region of image pixel values (float) for specified mip and slice
+            void GetSubImagePixelValues(AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<float> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
+
+            //! Retrieve a region of image pixel values (uint) for specified mip and slice
+            void GetSubImagePixelValues(AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<AZ::u32> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
+
+            //! Retrieve a region of image pixel values (int) for specified mip and slice
+            void GetSubImagePixelValues(AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<AZ::s32> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
```

Comment:
These should specify that both topLeft and bottomRight are inclusive. We have other APIs where the bottom corner is exclusive, so it's good to be super-explicit as to which way this one has chosen to go.

Reply:
I'm all for changing it to exclusive to be more consistent with our other APIs. I will add a comment to specify either way.

Reply:
Discussed offline, will change this to be exclusive and document it. If anyone else has a strong opinion for keeping it inclusive, please let me know!